src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,742 @@
+    bool private _isDirectOperationsOnly;
+
+    /// @notice Address of the project treasury which will receive the collateral tokens
+    address private _projectTreasury;
+
+    /// @dev    Storage gap for future upgrades.
+    uint[50] private __gap;
+
+    // -------------------------------------------------------------------------
+    // Modifiers
+
+    //--------------------------------------------------------------------------
+    // Initialization Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        // Initialize parent contracts
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode config data
+        (
+            address oracleAddress_,
+            address projectTreasury_,
+            address issuanceToken_,
+            address acceptedToken_,
+            uint buyFee_,
+            uint sellFee_,
+            uint maxSellFee_,
+            uint maxBuyFee_,
+            bool isDirectOperationsOnly_
+        ) = abi.decode(
+            configData_,
+            (address, address, address, address, uint, uint, uint, uint, bool)
+        );
+
+        // Set accepted token
+        _token = IERC20(acceptedToken_);
+
+        // Cache token decimals for collateral
+        _collateralTokenDecimals = IERC20Metadata(address(_token)).decimals();
+
+        if (
+            !ERC165Upgradeable(address(oracleAddress_)).supportsInterface(
+                type(IOraclePrice_v1).interfaceId
+            )
+        ) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_InvalidOracleInterface(
+            );
+        }
+        // Set oracle
+        _oracle = IOraclePrice_v1(oracleAddress_);
+
+        // Initialize base functionality (should handle token settings)
+        _setIssuanceToken(issuanceToken_);
+
+        // Set fees (checking max fees)
+        if (buyFee_ > maxBuyFee_) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                buyFee_, maxBuyFee_
+            );
+        }
+        if (sellFee_ > maxSellFee_) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                sellFee_, maxSellFee_
+            );
+        }
+        // Set project treasury
+        _setProjectTreasury(projectTreasury_);
+
+        // Set fees
+        _setBuyFee(buyFee_);
+        _setSellFee(sellFee_);
+
+        _setMaxBuyFee(maxBuyFee_);
+        _setMaxSellFee(maxSellFee_);
+
+        // Set direct operations only flag
+        setIsDirectOperationsOnly(isDirectOperationsOnly_);
+    }
+
+    // --------------------------------------------------------------------------
+    // Modifiers
+
+    /// @notice Modifier to check if direct operations are only allowed
+    modifier onlyDirectOperations() {
+        if (_isDirectOperationsOnly) {
+            revert
```

Comment:
Since this is only used to make buyFor and sellTo revert, I think renaming this to sth like "ThirdPartyOperationsEnabled" would make it clearer when reading the code

Reply:
modifier moved

Reply:
Thanks! 
Again, I'd consider renaming the modifier. As somebody looking into the code and making sense of it, the function header `buyFor() onlyModuleRole(x) onlyDirectOperations` reads as "the buyFor function can only be called by role X and and only when the directOperations flag is true" , while it actually does the opposite.

Reply:
I see your point, renamed to `ThirdPartyOperationsEnabled`